File: ppx/lua/git_checkout.lua
if not nyagos then
    print("This is a script for nyagos not lua.exe")
    os.exit()
end

local sb = nyagos.eval("git branch | peco")
if sb == "" then
    return nil
end

local gc = nyagos.exec("git checkout " .. sb)
if gc == 0 then
    function Get_si()
        local u, o = "", ""
        if sb ~= "" then
            local gitroot = nyagos.eval("git rev-parse --show-toplevel")
            local myrepo = string.gsub(arg[1], "\\", "/")
            if myrepo == gitroot then
                u = "*setcust _User:u_git_branch=" .. sb .. " %:"
            end
            if arg[2] ~= nil then
                o = "*execute C,*string i,oBranch=" .. sb
            end
            return {u, o}
        end
    end

    local si = Get_si()
    nyagos.exec("c:/bin/ppx/PPBW.EXE -c " .. si[1] .. si[2])

    os.exit()
end

File: nodejs/ppx/_git_checkout.js
'use strict';

// const fs = require('fs');
// const path = require('path');
const { execSync, exec } = require('child_process');

const ppx = 'c:\\bin\\ppx';

const bn = execSync('git branch | peco').toString().replace(/\n/,'');
if (!bn || bn.indexOf('*') == 0) { process.exit(); }

// レポジトリのルート、現在のブランチ名を取得
// const ditectRoot = ((resolve) => {
//   exec('git rev-parse --show-toplevel', (err, stdout) => {
//     if (err) {
//       console.error(err);
//       return;
//     }
//     resolve(stdout);
//   });
// });

exec(`${ppx}\\ppbw.exe -k git *stdout "%*input("checkout${bn}" -title:"git.." -select:9 -mode:e -k *completelist -file:%%'list'%%\\GITCOMMAND.TXT)"`, (err, stdout) => {
  if (err) {
    console.error(`${err}`);
    return;
  }
  console.log(stdout);
  // execSync(`c:\\bin\\ppx\\ppbw -c *execute C, *string i,oBRANCH=${stdout}`);
});
